### ISSUE TITLE
Publish long-form article: Recursive Self-Improvement Is Becoming an Engineering Problem

### ISSUE THREAD

**Issue by rmax** [rmax-ai/rmax-ai.github.io#78]
## Goal

Publish a **long-form rmax.ai article** based on the RSI research/draft developed separately:

> **Recursive Self-Improvement Is Becoming an Engineering Problem**  
> *From Turing’s learning machines to automated AI research — a grounded timeline of recursive self-improvement, where the field stands in 2026, and what builders should prepare for next.*

This should be implemented as a **long-form note using the existing `notes/<slug>/index.md` publishing model**.

Do **not** introduce a new article content type or route. The distinction here is editorial length and depth, not a different site primitive.

## Editorial thesis

RSI should be presented as a progression of increasingly closed improvement loops rather than a single speculative “intelligence explosion” event.

The core argument:

- We do **not** have public evidence of unrestricted runaway recursive self-improvement.
- We **do** have increasingly capable systems improving reasoning procedures, training signals, agent harnesses, algorithms, research workflows, and portions of AI R&D.
- The most important transition to watch is when AI materially accelerates the development of its successor models.
- The key engineering object is increasingly the **improvement loop**, not the standalone model.
- Preparing for RSI means learning to engineer, evaluate, secure, and measure closed improvement loops now.

### 1. What recursive self-improvement actually means
Define RSI carefully and distinguish it from:
- ordinary learning;
- fine-tuning;
- reflection;
- inference-time scaling;
- simple agent iteration.

Use the loop:

```
system
  ↓
propose improvements
  ↓
build / train / modify
  ↓
evaluate
  ↓
select
  ↓
deploy improved system
  ↓
system becomes better at proposing improvements
  ↺
```

Explain layers of self-improvement:
- behavior;
- agent architecture/harness;
- algorithms;
- research process;
- foundation-model development.

### 2. Timeline and milestones

Cover the progression:

- **1950 — Turing:** learning machines / child-machine framing.
- **1965 — I. J. Good:** intelligence explosion argument.
- **2003 — Gödel Machine:** provably beneficial self-modification.
- **2016–2018 — AlphaGo / AlphaZero:** self-play + search + objective-driven improvement.
- **2022 — STaR:** bootstrapped reasoning traces.
- **2023 — Reflexion:** improvement through persistent linguistic feedback without weight updates.
- **2024 — Self-Rewarding Language Models:** models contributing their own reward signal.
- **2024 — LLM² / DiscoPOP:** LLMs discovering training objectives.
- **2024 — AI Scientist:** automating larger portions of the scientific workflow.
- **2025 — AI Scientist-v2:** less scaffolded automated research.
- **2025 — AlphaEvolve:** LLM + evaluator + evolutionary search improving algorithms.
- **2025 — Darwin Gödel Machine:** agents modifying/evolving agent implementations.
- **2026 — Digital Red Queen:** open-ended adversarial program evolution.
- **2026 — AI Scientist published in Nature.**
- **2026 — Sakana RSI Lab:** RSI becomes an explicit research program.
- **2025/2026 — Frontier-lab safety frameworks:** automated AI R&D treated as a monitored capability.
- **September 2026 — OpenAI research acceleration:** “automated research intern” framing and increased AI-generated research work.

### 3. Where we actually are

Use a maturity ladder similar to:

| Stage | Improvement target | Status |
|---|---|---|
| 0 | Answers/task performance | Mature |
| 1 | Memory, prompts, strategies, skills | Common |
| 2 | Training data and reward signals | Demonstrated |
| 3 | Tools, agent architecture, code | Demonstrated |
| 4 | Algorithms and research workflows | Demonstrated |
| 5 | Large portions of AI R&D | Emerging |
| 6 | Successor frontier models with minimal human research input | Not publicly demonstrated |
| 7 | Sustained autonomous recursive model improvement | Not publicly demonstrated |

Be careful not to imply stages 6–7 have been achieved.

### 4. Why intelligence may not be the bottleneck

Discuss:
- evaluation quality / Goodharting;
- search cost;
- compute;
- experimental latency;
- security and control;
- research taste / allocation.

Emphasize that faster generation shifts scarcity toward **verification and selection**.

### 5. What to watch next

Use observable milestones rather than predicting an RSI date:

1. persistent automated research engineers;
2. self-improving harnesses becoming normal;
3. AI-designed evaluators co-evolving with AI-designed solutions;
4. research portfolios/populations replacing single-agent workflows;
5. measurable shortening of model-generation cycles.

### 6. How builders should prepare

Focus on practical engineering:

- engineer loops, not only agents;
- invest disproportionately in evals/verification;
- make experiments and knowledge machine-readable;
- separate proposal authority from deployment authority;
- sandbox self-modification;
- maintain provenance and immutable evidence;
- measure improvement velocity;
- track human interventions, cost per validated improvement, regression rate, evaluator disagreement, hypothesis-to-result latency, and percentage of R&D automated.

### 7. Ending

Land on this framing:

> The people best prepared for RSI will not be those who predicted the exact date. They will be those who learned how to measure, constrain, and engineer recursive improvement before the recursion became fast.

## Sources / grounding

Prefer primary papers and official lab material. At minimum include and link:

- Turing (1950), *Computing Machinery and Intelligence*  
  https://doi.org/10.1093/mind/LIX.236.433
- Good (1965), *Speculations Concerning the First Ultraintelligent Machine*  
  https://doi.org/10.1016/S0065-2458(08)60418-0
- Gödel Machine  
  https://arxiv.org/abs/cs/0309048
- AlphaZero  
  https://doi.org/10.1126/science.aar6404
- STaR  
  https://arxiv.org/abs/2203.14465
- Reflexion  
  https://papers.nips.cc/paper_files/paper/2023/hash/1b44b878bb782e6954cd888628510e90-Abstract-Conference.html
- Self-Rewarding Language Models  
  https://arxiv.org/abs/2401.10020
- LLM² / DiscoPOP  
  https://sakana.ai/llm-squared/
- AI Scientist  
  https://sakana.ai/ai-scientist/
- AI Scientist-v2  
  https://sakana.ai/ai-scientist-first-publication/
- Darwin Gödel Machine  
  https://arxiv.org/abs/2505.22954
- AlphaEvolve  
  https://deepmind.google/blog/alphaevolve-a-gemini-powered-coding-agent-for-designing-advanced-algorithms/
- Digital Red Queen  
  https://pub.sakana.ai/drq/
- AI Scientist — Nature  
  https://www.nature.com/articles/s41586-026-10265-5
- Sakana RSI Lab  
  https://sakana.ai/rsi-lab/
- METR task horizons  
  https://metr.org/time-horizons/
- OpenAI Preparedness Framework v2  
  https://cdn.openai.com/pdf/18a02b5d-6b67-4cec-ab64-68cdfbddebcd/preparedness-framework-v2.pdf
- OpenAI, *Research Acceleration: The View Inside OpenAI*  
  https://openai.com/index/research-acceleration-view-inside-openai/
- Anthropic Responsible Scaling Policy  
  https://www.anthropic.com/responsible-scaling-policy
- Google DeepMind Frontier Safety Framework  
  https://deepmind.google/frontier-safety/

## Editorial requirements

- Long-form note using the existing `notes/<slug>/index.md` structure.
- Serious, research-oriented rmax.ai voice.
- No AGI/RSI hype language presented as fact.
- Distinguish demonstrated results from lab claims, forecasts, and speculation.
- Prefer primary sources.
- Inline links throughout the article plus a visible references section.
- Preserve exact dates for recent 2026 milestones.
- Explain caveats around workshop acceptance, benchmarks, task-horizon extrapolation, and lab self-reports.
- Avoid implying that current self-improving agents autonomously retrain their underlying frontier foundation models.
- Include diagrams/tables where they materially clarify the argument.
- Follow existing rmax.ai visual/footer conventions.
- Validate the note frontmatter against `notes/schema.yaml`.

## Acceptance criteria

- [ ] Published as `notes/<slug>/index.md` using the existing note schema and site conventions.
- [ ] Timeline is fact-checked against primary sources.
- [ ] All factual claims about current frontier labs are sourced.
- [ ] “Where we are” clearly separates demonstrated capabilities from speculative future stages.
- [ ] Visible bibliography/reference section renders correctly.
- [ ] No tracking parameters in source links.
- [ ] Site build/checks pass.
- [ ] Page follows existing rmax.ai branding and footer conventions.

## Reference draft

The complete Markdown draft produced during research is preserved on the working branch for this issue:

- [`processed/rsi-timeline-what-comes-next-reference.md`](https://github.com/rmax-ai/rmax-ai.github.io/blob/issue-78-rsi-reference/processed/rsi-timeline-what-comes-next-reference.md)
- Branch: `issue-78-rsi-reference`
- Reference commit: `3d42c7ef3fe37a3ee87369e91fec6df217363581`

Use this as the **editorial/reference draft**, not as a requirement to publish it verbatim. Fact-check, adapt to the site's article format, and preserve the grounded source trail.
